app/lib/status_reach_finder.rb

```diff
@@ -20,7 +20,12 @@ def reached_account_inboxes
     # directly, we assume all interactions are with the original one
 
     if @status.reblog?
-      []
+      reblogged_status_account = @status.reblog.account
+      if !reblogged_status_account.local? && reblogged_status_account.activitypub? && !reblogged_status_account.following?(@status.account)
+        [reblogged_status_account.inbox_url]
+      else
+        []
+      end
     else
       Account.where(id: reached_account_ids).inboxes
     end
```

Comment:
I think keeping the original `Account.where(...).inboxes` code makes more sense, instead of checking for `.local?` and `.activitypub?` manually. Instead, we should move the `if @status.reblog?` condition down to `reached_account_ids`, so that we return either `[reblog_of_account_id]` if the status is a reblog, or `[replied_to_account_id, mentioned_account_ids, ...]` etc otherwise

Reply:
(Also, the `!following?` condition isn't needed anymore, since the `uniq` on line 13 takes care of that)

Reply:
I think it should read:

```ruby
[@status.reblog.account.preferred_inbox_url.presence].compact
```

Indeed, local and OStatus accounts won't have an `inbox_url` or `shared_inbox_url`, it will return `''`, so `#presence` will turn that into a `null`, and `compact` will take care of that.

Reply:
As I mentioned in my comment, while `[@status.reblog.account.preferred_inbox_url.presence].compact` would *work*, I think keeping the logic in this method the same for both reblog and non-reblog statuses and modifying `reached_account_ids` makes more sense and keeps the code cleaner.

Reply:
Yeah, that works too. I read your comment a bit fast and I think your solution is indeed slightly more elegant.

Reply:
Moved the process of getting the inbox url of the poster of original post to `reached_account_ids`.